beacon_node/execution_layer/src/engines.rs

```diff
@@ -296,6 +314,7 @@ impl<T: EngineApi> Engines<T> {
 
         for engine in &self.engines {
             let engine_synced = *engine.state.read().await == EngineState::Synced;
+            let engine_auth_failed = *engine.state.read().await == EngineState::AuthFailed;
```

Comment:
I'm a little peeved here because we get different errors for the 2 invocations. The first invocation will always return `Offline` error and the second error will be the actual error if it exists (e.g. Auth error). This is mostly just a logging issue that is relevant in tests but just wanted to point it out.
for e.g. running the `finds_valid_terminal_block_hash` test with auth disabled results in the following error.
```
thread 'test::finds_valid_terminal_block_hash' panicked at 'called `Result::unwrap()` on an `Err` value: EngineErrors([Offline { id: "http://127.0.0.1:36749/" }, Auth { id: "http://127.0.0.1:36749/" }])'
```

Reply:
Oh yeah, that is a bit annoying. I think it's because we don't upcheck the nodes before we start tests. I think that would get rid of the first `Offline` error.

Regarding the two `read`s here, maybe we could be reduce them to a single `read` with:

```rust
let (engine_synced, engine_auth_failed) = {
  let state = engine.state.read().await;
  (state == EngineState::Synced, state == EngineState::AuthFailed)
};
```

The surrounding braces ensure we don't hold the read-lock.